### PR TITLE
types for contents + progressions

### PR DESCRIPTION
**Pull request by chrisdugne**
**Detailed purpose of the PR**

**Result and observation**

- [ ] **Breaking changes ?**  
       If checked, what have you broken ?

- [ ] **Extra lib ?**
      If checked, Which extra lib did you add ? (name, purpose, link ...).

**Testing Strategy**

- [x] Already covered by tests
- [ ] Manual testing
- [ ] Unit testing

### CODE REVIEW

File: packages/@coorpacademy-player-services/src/content.js [hunk 1]
```diff
@@ -29,14 +41,18 @@ const getNbSlides = fixtures => (contentRef, engineRef, version) => {
   return -1;
 };
 
-const getInfo = fixtures => (contentRef, engineRef, version) => {
+const getInfo = (fixtures: Fixtures) => (
```
Comment (VincentCATILLON) [inline packages/@coorpacademy-player-services/src/content.js]: Can you add return type ?

File: packages/@coorpacademy-player-services/src/content.js [hunk 2]
```diff
@@ -1,13 +1,25 @@
...
+const find = (fixtures: Fixtures) => (
+  type: string,
+  ref: string
+): Promise<Chapter | Level | Slide> => {
   const {findContent} = fixtures;
   return findContent(type, ref);
 };
 
-const getNbSlides = fixtures => (contentRef, engineRef, version) => {
+const getNbSlides = (fixtures: Fixtures) => (
```
Comment (VincentCATILLON) [inline packages/@coorpacademy-player-services/src/content.js]: Can you add return type ?

File: packages/@coorpacademy-player-services/src/content.js [hunk 3]
```diff
@@ -29,14 +41,18 @@ const getNbSlides = fixtures => (contentRef, engineRef, version) => {
...
+  contentRef: string,
+  engineRef: string,
+  version: string
+): {nbSlides: number} => {
   const nbSlides = getNbSlides(fixtures)(contentRef, engineRef, version);
   return {nbSlides};
 };
 
-const Content = fixtures => ({
+const ContentService = (fixtures: Fixtures) => ({
```
Comment (VincentCATILLON) [inline packages/@coorpacademy-player-services/src/content.js]: Can you add return type ?

File: packages/@coorpacademy-player-services/src/progressions.js [hunk 4]
```diff
@@ -172,7 +283,7 @@ const markResourceAsViewed = fixtures => async (progressionId, payload) => {
   return addActionAndSaveProgression(fixtures)(progression, action);
 };
 
-const Progressions = fixtures => ({
+const Progressions = (fixtures: Fixtures) => ({
```
Comment (VincentCATILLON) [inline packages/@coorpacademy-player-services/src/progressions.js]: Can you add return type ?

File: packages/@coorpacademy-player-services/src/test/fixtures/content.js [hunk 5]
```diff
@@ -1,27 +1,32 @@
...
 import {findById as findChapterById} from './chapters';
 import {findById as findLevelById} from './levels';
 import {findById as findSlideById} from './slides';
 
 // eslint-disable-next-line import/prefer-default-export
-export const find = (type, ref) => {
+export const find = (type: string, ref: string): Promise<Chapter | Level | Slide> => {
   switch (type) {
-    case 'chapter': {
+    case CONTENT_TYPE.CHAPTER: {
```
Comment (VincentCATILLON) [inline packages/@coorpacademy-player-services/src/test/fixtures/content.js]: 🚀

File: packages/@coorpacademy-progression-engine/src/types.js [hunk 6]
```diff
@@ -8,9 +8,22 @@ export type Step = {
...
+export type FAILURE = 'failure';
+export type SUCCESS = 'success';
+export type ContentType = CHAPTER | LEVEL | SLIDE | NODE | FAILURE | SUCCESS;
+
+export type ContentSlide = {|
+  type: SLIDE,
+  ref: string
+|};
+
 export type GenericContent = {
```
Comment (VincentCATILLON) [inline packages/@coorpacademy-progression-engine/src/types.js]: If you can, and if it's mandatory for the mobile starts needs, please use strict for these types. Else, don't take care of this comment :)

File: packages/@coorpacademy-player-services/src/content.js (hunk 1 above)
Comment (chrisdugne) [reply packages/@coorpacademy-player-services/src/content.js]: ✅

File: packages/@coorpacademy-player-services/src/content.js (hunk 2 above)
Comment (chrisdugne) [reply packages/@coorpacademy-player-services/src/content.js]: ✅

File: packages/@coorpacademy-player-services/src/content.js (hunk 3 above)
Comment (chrisdugne) [reply packages/@coorpacademy-player-services/src/content.js]: ✅

File: packages/@coorpacademy-player-services/src/progressions.js (hunk 4 above)
Comment (chrisdugne) [reply packages/@coorpacademy-player-services/src/progressions.js]: ✅

File: packages/@coorpacademy-progression-engine/src/types.js (hunk 6 above)
Comment (chrisdugne) [reply packages/@coorpacademy-progression-engine/src/types.js]: ✅  --> some types are to intricated (State, Slide)
for many it's ok

**Pull request merged**